hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTimelineArchiveLog.java

```diff
@@ -329,7 +329,8 @@ private void writeToFile(Schema wrapperSchema, List<IndexedRecord> records) thro
     if (records.size() > 0) {
       Map<HeaderMetadataType, String> header = new HashMap<>();
       header.put(HoodieLogBlock.HeaderMetadataType.SCHEMA, wrapperSchema.toString());
-      HoodieAvroDataBlock block = new HoodieAvroDataBlock(records, header);
+      final String keyField = table.getMetaClient().getTableConfig().getRecordKeyFieldProp();
```

Comment:
what if its an older table, where this field is not present?

Reply:
Probably the upgrade process will take care of it by adding the required entires to the hoodie.properties?

Reply:
Made the getTableConfig().getRecordKeyFieldProp() to return the default old key field if the key property is not available. Hope this addresses the old tables.

Reply:
with 0.10.0, we blow out the metadata and re-build right. Or are you asking, if someone uses hudi-0.10.0 lib to read a old table w/o triggering a single write to hudi table? bcoz, if one write is triggered, metadata will be fully bootstrapped and I assume we add all these props to table props.